src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/location/importer/LocationsSpreadsheet.kt

```diff
@@ -36,11 +43,21 @@ class LocationsSpreadsheet(private val spreadsheet: Workbook) : Closeable {
     fun getRowsFrom(tab: Tab): List<Row> = spreadsheet.getSheet(tab.label).drop(COLUMN_HEADINGS).filterNot { it.getCell(1)?.stringCellValue.isNullOrBlank() }
 
     fun mapToLocation(cells: List<Cell>): Location {
+        val locationType = LocationType.map(cells[TYPE].stringCellValue.toUpperCase().trim())
+                ?: throw IllegalArgumentException("Unsupported location type: " + cells[TYPE].stringCellValue)
+
+        val agency = cells[AGENCY].stringCellValue.toUpperCase().trim().takeUnless { it == "" }
```

Comment:
```suggestion
        val agency = cells[AGENCY].stringCellValue.toUpperCase().trim().takeUnless { it == "" }!!
```

if you are finding the code a bit noisy you could force a NullPointerException

Reply:
Only makes sense if you don't care what exception you get

Reply:
Hmm don't think appropriate but does raise the question of the exception as could be an NPE instead.